### PR TITLE
Migrated the Backend APP Gateway module from the cnp-aks-rbac-pipelines repository.

### PR DESCRIPTION
**Pull request by gbstan92**
### JIRA link (if applicable) ###

[AB#833](https://dev.azure.com/hmcts/90472b63-dc8d-4ae6-8819-a32bd211e914/_workitems/edit/833)

### Change description ###

Migrated the Backend APP Gateway module from the cnp-aks-rbac-pipelines repository. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: azure_pipeline.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ stages:
       - template: pipeline-templates/terraform-plan.yaml
         parameters:
           environment: 'sbox'
-          component: 'cftapps_cluster_lb'
+          component: 'cftapps_cluster_lb_frontend'
```
Comment (timja) [inline azure_pipeline.yaml]: This is used for the state file key so you won’t want to rename if unless you rename the state file in all envs

**Comment by timja** [comment]
Can you not name your PR as initial commit :p move your description up there

File: azure_pipeline.yaml (hunk 1 above)
Comment (gbstan92) [reply azure_pipeline.yaml]: I realised this and reverted the change for the module. I left the naming convention for it as it was and only added "_backend" to the new module.

File: environments/ithc/ithc.tfvars [hunk 2]
```diff
@@ -6,6 +6,9 @@ enable_ssl                 = true
 ssl_mode                   = "AzureKeyVault"
 certificate_key_vault_name = "cftapps-ithc"
 
+app_gw_private_ip_address = "10.10.36.121"
+vault_name                = "cftapps-ithc"
```
Comment (timja) [inline environments/ithc/ithc.tfvars]: the vault name is already there on line 7

Comment (gbstan92) [reply environments/ithc/ithc.tfvars]: Done!

**Comment by gbstan92** [comment]
We've only migrated sandbox so far to ensure deployment is succesfull. Once the build goes green on master I'll proceed migrating the rest of the envs.

**Pull request merged**